libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -851,6 +865,60 @@ AP_GPS_UBLOX::_parse_gps(void)
         state.horizontal_accuracy = 0;
 #endif
         break;
+
+    case MSG_NAV_RELPOSNED: {
+        Debug("MSG_NAV_RELPOSNED RTK status Received");
+
+        state.rtk_age_ms    = 0xFFFFFFFF;
+
+    	state.rtk_baseline_x_mm = _buffer.rtk.rel_pos_n_cm*10+_buffer.rtk.rel_pos_hp_n_mm/10.0;
+    	state.rtk_baseline_y_mm = _buffer.rtk.rel_pos_e_cm*10+_buffer.rtk.rel_pos_hp_e_mm/10.0;
+    	state.rtk_baseline_z_mm = _buffer.rtk.rel_pos_d_cm*10+_buffer.rtk.rel_pos_hp_d_mm/10.0;
+
+    	int32_t acc_n_mm = _buffer.rtk.acc_n_mm;
+    	int32_t acc_e_mm = _buffer.rtk.acc_e_mm;
+    	int32_t acc_d_mm = _buffer.rtk.acc_d_mm;
+
+    	// rms of error terms
+    	double rtk_accuracy = acc_n_mm*acc_n_mm + acc_e_mm*acc_e_mm + acc_d_mm*acc_d_mm;
```

Comment:
There's some serious type mismatching going on here, you are doing all your math as int's and then storing into a double, but you've already risked overflowing in the pure integer math before converting to a double. You then pass this to `safe_sqrt()` which is going to give you a float result, so you might as well have stored this in a float in the first place, particularly because this is just a monitor, and you shouldn't ever appreciate the resolution difference here, since this is going to be dominated by casting into a unsigned type.

Reply:
Float won't overflow, and won't lose precision unless you have massive values. I'd prefer this to be float, because on most devices you just forced it into a very slow software path to do the double calculation. Since it's just a monitoring, and on something we shouldn't notice the loss on anyways) I'd like to see this as float.